lib/lookup/src/lookup_v2/mod.rs

```diff
@@ -220,3 +261,57 @@ impl<'a> From<BorrowedSegment<'a>> for OwnedSegment {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn owned_path_display() {
+        let test_cases = [
+            ("", ".<invalid>"),
+            (".", "."),
+            ("]", ".<invalid>"),
+            ("]foo", ".<invalid>"),
+            ("..", ".<invalid>"),
+            ("...", ".<invalid>"),
+            ("f", ".f"),
```

Comment:
These are the same thing. A leading `.` means it's part of the event vs a local. This isn't used in VRL today, but it eventually will be and should match.